CMakeLists.txt

```diff
@@ -87,34 +88,36 @@ endif()
 
 find_package(Python3 ${LLVM_MINIMUM_PYTHON_VERSION} REQUIRED COMPONENTS Interpreter Development)
 
-# On MacOS, find_package for Java and JNI doesn't work if JAVA_INCLUDE_PATH
-# and JAVA_INCLUDE_PATH2 are not set. Setting them requires JAVA_HOME to be
-# known. If the JAVA_HOME env var isn't set, we try to figure it out by running
-# java and looking for the java.home property.
-if (APPLE)
-  set(JAVA_HOME $ENV{JAVA_HOME})
-  if ("${JAVA_HOME}" STREQUAL "")
-    execute_process(
-      COMMAND bash -c "java -XshowSettings:properties -version 2>&1 | \
-                       sed -n 's/[ \t]*java.home = //p'"
-      OUTPUT_STRIP_TRAILING_WHITESPACE
-      OUTPUT_VARIABLE JAVA_HOME)
-  endif()
-  if (NOT "${JAVA_HOME}" STREQUAL "")
-    SET(JAVA_INCLUDE_PATH  "${JAVA_HOME}/include")
-    SET(JAVA_INCLUDE_PATH2 "${JAVA_HOME}/include/darwin")
+if(ONNX_MLIR_ENABLE_JAVA)
+  # On MacOS, find_package for Java and JNI doesn't work if JAVA_INCLUDE_PATH
+  # and JAVA_INCLUDE_PATH2 are not set. Setting them requires JAVA_HOME to be
+  # known. If the JAVA_HOME env var isn't set, we try to figure it out by running
+  # java and looking for the java.home property.
+  if (APPLE)
+    set(JAVA_HOME $ENV{JAVA_HOME})
+    if ("${JAVA_HOME}" STREQUAL "")
+      execute_process(
+        COMMAND bash -c "java -XshowSettings:properties -version 2>&1 | \
+                         sed -n 's/[ \t]*java.home = //p'"
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        OUTPUT_VARIABLE JAVA_HOME)
+    endif()
+    if (NOT "${JAVA_HOME}" STREQUAL "")
+      SET(JAVA_INCLUDE_PATH  "${JAVA_HOME}/include")
+      SET(JAVA_INCLUDE_PATH2 "${JAVA_HOME}/include/darwin")
+    endif()
   endif()
-endif()
 
-# Don't require AWT to allow headless JDK to reduce docker image size
-set(JAVA_AWT_INCLUDE_PATH headless)
-set(JAVA_AWT_LIBRARY headless)
-find_package(Java COMPONENTS Development)
-find_package(JNI)
+  # Don't require AWT to allow headless JDK to reduce docker image size
+  set(JAVA_AWT_INCLUDE_PATH headless)
+  set(JAVA_AWT_LIBRARY headless)
+  find_package(Java COMPONENTS Development)
+  find_package(JNI)
 
-if (Java_Development_FOUND AND JNI_FOUND)
-  set(ONNX_MLIR_ENABLE_JNI TRUE)
-  include(UseJava)
+  if (Java_Development_FOUND AND JNI_FOUND)
+    set(ONNX_MLIR_ENABLE_JNI TRUE)
+    include(UseJava)
+  endif()
 else()
   set(ONNX_MLIR_ENABLE_JNI FALSE)
   message(WARNING "Java Development component or JNI not found, JNI targets will not work")
```

Comment:
This message is for the case when `ONNX_MLIR_ENABLE_JAVA` is ON but Java Development component or JNI is not found, not when `ONNX_MLIR_ENABLE_JAVA` is OFF.

Reply:
Oops, sorry about that! I restored the `"Java Development component or JNI not found, JNI targets will not work"` message to its proper place and added a different message (`"Java explicitly disabled with the ONNX_MLIR_ENABLE_JAVA option, JNI targets will not work"`) when the new option is used to disable Java explicitly.